### PR TITLE
xorg-cf-files: add xorg-cf-files/1.0.7 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **xorg-cf-files/1.0.7**

This recipe will need an exception in conan-io/hooks as it packages only scripts.

- [x] Depends on #6719

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xorg-cf-files/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
+        if self.settings.compiler == "Visual Studio":
+            self.build_requires("automake/1.16.3")
+
+    def configure(self):
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def source(self):
```
Comment (madebr) [inline recipes/xorg-cf-files/all/conanfile.py:45]:
```suggestion
        del self.settings.compiler.libcxx

    def validate(self):
        if tools.is_apple_os(self.settings.os):
            raise ConanInvalidConfiguration("This recipe does not support Apple operating systems.")

    def source(self):
```

File: recipes/xorg-cf-files/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
```
Comment (madebr) [inline recipes/xorg-cf-files/all/conanfile.py:1]:
```suggestion
from conans import ConanFile, tools, AutoToolsBuildEnvironment
from conans.errors import ConanInvalidConfiguration
```

File: recipes/xorg-cf-files/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+
+class XorgCfFilesConan(ConanFile):
+    name = "xorg-cf-files"
+    description = "Imake configuration files & templates"
+    topics = ("conan", "imake", "xorg", "template", "configuration", "obsolete")
+    license = "MIT"
+    homepage = "https://gitlab.freedesktop.org/xorg/util/cf"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler"
```
Comment (madebr) [inline recipes/xorg-cf-files/all/conanfile.py:16]:
```suggestion
    settings = "os"
```

File: recipes/xorg-cf-files/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+
+class XorgCfFilesConan(ConanFile):
+    name = "xorg-cf-files"
+    description = "Imake configuration files & templates"
+    topics = ("conan", "imake", "xorg", "template", "configuration", "obsolete")
+    license = "MIT"
+    homepage = "https://gitlab.freedesktop.org/xorg/util/cf"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os"  # FIXME: can be removed once c3i is able to test multiple os'es from one common package
```
Comment (madebr) [inline recipes/xorg-cf-files/all/conanfile.py:16]:
```suggestion
    settings = "os", "compiler"
```

File: recipes/xorg-cf-files/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+    def configure(self):
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def validate(self):
+        if tools.is_apple_os(self.settings.os):
+            raise ConanInvalidConfiguration("This recipe does not support Apple operating systems.")
+
+    def source(self):
```
Comment (madebr) [inline recipes/xorg-cf-files/all/conanfile.py:50]:
```suggestion
            raise ConanInvalidConfiguration("This recipe does not support Apple operating systems.")

    def package_id(self):
        del self.info.settings.compiler
        # self.info.settings.os  # FIXME: can be removed once c3i is able to test multiple os'es from one common package

    def source(self):
```

**Comment by madebr** [comment]
Depends on https://github.com/conan-io/hooks/pull/331

File: recipes/xorg-cf-files/all/conandata.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+sources:
+  "1.0.8":
+    url: "https://www.x.org/pub/individual/util/xorg-cf-files-1.0.6.tar.gz"
```
Comment (jgsogo) [inline recipes/xorg-cf-files/all/conandata.yml:3]: Version mismatch? Is it 1.0.6 or 1.0.8?

Comment (jgsogo) [reply recipes/xorg-cf-files/all/conandata.yml]: LOL 🤣

**Review by uilianries — APPROVED** [review]

File: recipes/xorg-cf-files/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+import contextlib
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class XorgCfFilesConan(ConanFile):
+    name = "xorg-cf-files"
+    description = "Imake configuration files & templates"
+    topics = ("conan", "imake", "xorg", "template", "configuration", "obsolete")
```
Comment (sparik) [inline recipes/xorg-cf-files/all/conanfile.py:12]: conan can be removed from topics

**Review by SSE4 — APPROVED** [review]

**Pull request merged**